src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs

```diff
@@ -79,20 +78,27 @@ internal class TextBufferIssueTracker : IIssueTracker
 
             this.document = document;
             this.FilePath = document.FilePath;
-
-            this.ProjectItem = dte.Solution.FindProjectItem(this.FilePath);
             this.charset = document.Encoding.WebName;
 
-            // Bug #676: https://github.com/SonarSource/sonarlint-visualstudio/issues/676
-            // It's possible to have a ProjectItem that doesn't have a ContainingProject
-            // e.g. files under the "External Dependencies" project folder in the Solution Explorer
-            var projectName = this.ProjectItem?.ContainingProject.Name ?? "{none}";
-            this.Factory = new SnapshotFactory(new IssuesSnapshot(projectName, this.FilePath, 0,
-                new List<IssueMarker>()));
+            this.Factory = new SnapshotFactory(new IssuesSnapshot(ProjectName, FilePath, 0, new List<IssueMarker>()));
 
             document.FileActionOccurred += SafeOnFileActionOccurred;
         }
 
+        private string ProjectName
```

Comment:
I think this should be a `GetProjectName` method rather than a property. I wouldn't expect a property to do this much processing when it's called. Also, this code is actually thread-sensitive i.e. we should be calling the `dte` on the UI thread., which I wouldn't expect to be done in property.